src/Utilities/ProtocolHelpers.hpp

```diff
@@ -60,7 +60,11 @@ struct AssertConformsToImpl : std::true_type {
       "and the protocol is listed as the second template parameter. "
       "Have you forgotten to (publicly) inherit the type from "
       "tt::ConformsTo<Protocol>?");
-  using test = typename Protocol::template test<ConformingType>;
+  // The `decltype` makes sure the conformance test is not optimized out by
```

Comment:
I am surprised if this ever worked as a the type alias you deleted does not require a definition of `Protocol::test`. I would change the comment to `// Implicitly instantiate Protocol::test in order to test conformance`